File: src/platform/time/engine_time.cpp
#include "sani/platform/time/engine_time.hpp"

namespace sani {

	EngineTime::EngineTime(const float64 totalTime, const float64 frameTime) : totalTime(totalTime),
																			   frameTime(frameTime) {
	}

	float64 EngineTime::getTotalTime() const {
		return totalTime;
	}
	float64 EngineTime::getFrameTime() const {
		return frameTime;
	}

	EngineTime::~EngineTime() {
	}
}

File: vs2013/cs/SaNi.Mono/Engine/EngineTime.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SaNi.Mono.Engine
{
    public struct EngineTime
    {
        public readonly TimeSpan TotalTime;
        public readonly TimeSpan FrameTime;

        public EngineTime(float totalTime, float frameTime)
        {
            TotalTime = TimeSpan.FromMilliseconds(totalTime);
            FrameTime = TimeSpan.FromMilliseconds(frameTime);
        }
    }
}
